spock-core/src/main/java/org/spockframework/runtime/extension/builtin/UnrollExtension.java

```diff
@@ -15,49 +15,88 @@
 package org.spockframework.runtime.extension.builtin;
 
 import org.spockframework.runtime.DataVariablesIterationNameProvider;
-import org.spockframework.runtime.GroovyRuntimeUtil;
-import org.spockframework.runtime.extension.AbstractAnnotationDrivenExtension;
-import org.spockframework.runtime.model.FeatureInfo;
-import org.spockframework.runtime.model.IterationInfo;
-import org.spockframework.runtime.model.NameProvider;
-import org.spockframework.runtime.model.SpecInfo;
+import org.spockframework.runtime.InvalidSpecException;
+import org.spockframework.runtime.extension.AbstractGlobalExtension;
+import org.spockframework.runtime.model.*;
 
+import spock.lang.Rollup;
 import spock.lang.Unroll;
 
-public class UnrollExtension extends AbstractAnnotationDrivenExtension<Unroll> {
-  private final String globalUnrollPattern;
-
-  public UnrollExtension() {
-    String globalUnrollPattern = System.getProperty("spock.globalUnrollPattern");
-    this.globalUnrollPattern = GroovyRuntimeUtil.isTruthy(globalUnrollPattern) ? globalUnrollPattern : null;
-  }
+public class UnrollExtension extends AbstractGlobalExtension {
+  private UnrollConfiguration unrollConfiguration;
 
   @Override
-  public void visitSpecAnnotation(Unroll unroll, SpecInfo spec) {
-    for (FeatureInfo feature : spec.getFeatures()) {
-      if (feature.isParameterized()) {
-        visitFeatureAnnotation(unroll, feature);
-      }
+  public void visitSpec(SpecInfo spec) {
+    Unroll unroll = spec.getAnnotation(Unroll.class);
+    boolean unrollSpec = unroll != null;
+    boolean rollupSpec = spec.getAnnotation(Rollup.class) != null;
+
+    if (unrollSpec && rollupSpec) {
+      throw new InvalidSpecException("@Unroll and @Rollup must not be used on the same spec: " + spec.getName());
     }
+
+    boolean doUnrollSpec;
+    String specUnrollPattern;
+    if (unrollSpec) {
+      doUnrollSpec = true;
+      specUnrollPattern = unroll.value();
+    } else if (rollupSpec) {
+      doUnrollSpec = false;
+      specUnrollPattern = "";
+    } else {
+      doUnrollSpec = unrollConfiguration.enabledByDefault;
+      specUnrollPattern = "";
+    }
+
+    spec
```

Comment:
Did you add a Spec that checks that inheritance works as expected?

What are the expected results in these scenarios

```
Root

@Unroll
Base extends Root

A extends Base

@Rollup
B extends A

C extends B

@Unroll
D extends C
```

other way round

```
Root

@Rollup
Base extends Root

A extends Base

@Unroll
B extends A

C extends B

@Rollup
D extends C
```

Reply:
I think I did not.

I added tests that you cannot have `@Unroll` and `@Rollup` on the same feature or spec and that you can overwrite on feature level what you declared on spec level and on feature and spec level what you configured in the configuration script.
But I think I didn't do inheritance tests.

What do you think how it should behave?
How die `@Unroll` behave up to now?

Reply:
Ok, in 1.3 the `@Unroll` on the super-spec makes the method unroll, but the pattern from that annotation is not used, maybe we can even improve on that.

Reply:
Ah, no, I tested this accidentally with the `spock-global-unroll` of @szpak on the classpath.
Which actually does not work on the features of super specs at it seems.
So with the proper test setup it is like how the annotation is defined, it is not inheritable.
The annotations only work on the spec they are attached to, not on sub-sepcs.
And the current implementation did exactly the same.
Do you think this should be different?

Reply:
Hm, well, it seems to maybe not do exactly the same, still investigating.
But also still the question how we want it.

Reply:
Ok, at least the strange behavior is fixed

Reply:
@leonard84 What are your preferences in that topic?

Reply:
Non inheritable behavior is ok, IMHO